controllers/machinedeployment_sync.go

```diff
@@ -436,6 +436,11 @@ func (r *MachineDeploymentReconciler) scaleMachineSetOperation(ctx context.Conte
 		*(ms.Spec.Replicas) = newScale
 		mdutil.SetReplicasAnnotations(ms, *(deployment.Spec.Replicas), *(deployment.Spec.Replicas)+mdutil.MaxSurge(*deployment))
 
+		// Update the delete policy for scale down operation
+		if deployment.Spec.Strategy.RollingUpdate.DeletePolicy != nil {
+			ms.Spec.DeletePolicy = *deployment.Spec.Strategy.RollingUpdate.DeletePolicy
```

Comment:
is it intentional to overwrite the MachineSet delete policy? what if the user explicitly sets a DeletePolicy in the machine set spec and in the machine deployment spec?

Reply:
My understanding was that if a MachineDeployment adopts a machine set, which is the scenario where there could potentially be a conflict, the deployment's settings takes precedence over that of the machineset. I Was looking at a similar setting `MinReadySeconds`, but that's only being set in the [new machine set](https://github.com/kubernetes-sigs/cluster-api/blob/ed580e2b92e3ebfbece20d31407c24c443432da4/controllers/machinedeployment_sync.go#L117).

Reply:
+1, I think we should likely avoid modifying existing MachineSets, and let any delta between the MachineDeployment configuration and MachineSet configuration be handled through the rollout process.

It might be good to verify that DeletePolicy is taken into account when determining if a change needs to be rolled out in the current implementation, though.

Reply:
So just to confirm, we just propagate the delete policy to the machine set that is going to be used as part of the current rollout, and leave the old ones as it is. Any scale down on an old ms will use its own delete policy. Is my understanding correct?